bundles/org.openhab.binding.homekit/src/main/java/org/openhab/binding/homekit/internal/handler/HomekitAccessoryHandler.java

```diff
@@ -0,0 +1,983 @@
+                                            .withDefaultTags(channelType.getTags()).withKind(channelType.getKind())
+                                            .withProperties(chanDef.getProperties()).withType(channelType.getUID());
+                                    Optional.ofNullable(chanDef.getLabel()).ifPresent(builder::withLabel);
+                                    Optional.ofNullable(chanDef.getDescription()).ifPresent(builder::withDescription);
+                                    Channel channel = builder.build();
+                                    channels.add(channel);
+
+                                    logger.trace(
+                                            "+++++Channel acceptedItemType:{}, defaultTags:{}, description:{}, kind:{}, label:{}, properties:{}, uid:{}",
+                                            channel.getAcceptedItemType(), channel.getDefaultTags(),
+                                            channel.getDescription(), channel.getKind(), channel.getLabel(),
+                                            channel.getProperties(), channel.getUID());
+
+                                }
+                            }
+                        });
+                    }
+                });
+
+        lightModelFinalize(accessory, channels);
+        stopMoveFinalize(accessory, channels);
+        eventingFinalize(accessory, channels);
+
+        String oldLabel = thing.getLabel();
+        String newLabel = oldLabel == null || oldLabel.isEmpty() ? accessory.getAccessoryInstanceLabel() : null;
+        List<Channel> newChannels = !channels.isEmpty() ? channels : null;
+        Map<String, String> newProperties = !properties.isEmpty() ? properties : null;
+        SemanticTag newTag = accessory.getSemanticEquipmentTag();
+
+        if (newLabel != null || newChannels != null || newProperties != null || newTag != null) {
+            ThingBuilder builder = editThing().withProperties(properties).withChannels(channels);
+            Optional.ofNullable(newLabel).ifPresent(builder::withLabel);
+            Optional.ofNullable(newChannels).ifPresent(builder::withChannels);
+            Optional.ofNullable(newProperties).ifPresent(builder::withProperties);
+            Optional.ofNullable(newTag).ifPresent(builder::withSemanticEquipmentTag);
+
+            updateThing(builder.build());
+            logger.debug("Updated thing {} channels, {} properties, label {}, tag {}", channels.size(),
+                    properties.size(), newLabel, newTag);
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        Channel channel = thing.getChannel(channelUID);
+        if (channel == null) {
+            logger.warn("Received command '{}' for unknown channel '{}'", command, channelUID);
+            return;
+        }
+        if (command == RefreshType.REFRESH) {
+            return;
```

Comment:
Can this be implemented? I noticed yesterday when I linked an item to a `window#position-target` channel, it stayed `UNDEF` and I had to disable/enable the Thing for the state to update.